src/Thumbnails/Thumbnails.js

```diff
@@ -21,72 +20,70 @@ const useStyles = makeStyles((theme) => ({
-//     title: 'Ralph',
-//     author: 'author',
-//    },
-//     {
-//     img: 'https://erinwritesnow.files.wordpress.com/2017/01/cute-happy-older-lady.jpg',
-//     title: 'Doris',
-//     author: 'author',
-//     },
-//    {
-//     img: 'https://erinwritesnow.files.wordpress.com/2017/01/cute-happy-older-lady.jpg',
-//     title: 'Doris',
-//     author: 'author',
-//     },
-//     {
-//     img: 'https://previews.123rf.com/images/natakot/natakot1810/natakot181001281/110094284-happy-old-man-showing-thumbs-up-working-out-with-a-dumbbell.jpg',
-//     title: 'Ralph',
-//     author: 'author',
-//    },
-//  ];
  
-export default function SingleLineGridList({availableResidents, semesterAvailable}) {
+export default function Thumbnails({availableResidents, allResidents, selectResident, semesterAvailable}) {
   const classes = useStyles();
-
-  return (
-    <div className={classes.root}>
-      <GridList className={classes.gridList} cols={2.5}>
-        {availableResidents.map((resident) => (
-          <GridListTile key={resident.photo}>
-            <img src={resident.photo} alt={resident.name} />
-            <GridListTileBar
-              title={resident.name}
-              classes={{
-                root: classes.titleBar,
-                title: classes.title,
-              }}
-              actionIcon={
-                <IconButton aria-label={`star ${resident.name}`}>
-                  {/* <StarBorderIcon className={classes.title} /> */}
-                </IconButton>
-              }
-            />
-          </GridListTile>
-        ))}
-      </GridList>
-    </div>
-  );
+ 
+  const handleClick = (resident) => { 
+    // let clickedResident = event.target.value
+    console.log(resident)
+    selectResident(resident)
+  }
+  if (availableResidents.length) {
+    return (
+      <div className={classes.root}>
+        <GridList className={classes.gridList} cols={2.5}>
+          {availableResidents.map((resident, index) => (
+            <GridListTile key={index} value={resident.id} onClick={() => handleClick(resident)}>
+              <img src={resident.photo} key={resident.id} id={resident.id} alt={resident.name} />
+              <GridListTileBar
+                title={resident.name}
+                classes={{
+                  root: classes.titleBar,
+                  title: classes.title,
+                }}
+                actionIcon={
+                  <IconButton aria-label={`star ${resident.name}`} >
+                    <StarBorderIcon className={classes.title}/>
+                  </IconButton>
+                }
+              />
+            </GridListTile>
+          ))}
+        </GridList>
+      </div>
+    );
+  } else {
+    return (
+      <div className={classes.root}>
+        <GridList className={classes.gridList} cols={2.5}>
+          {allResidents.map((resident, index) => (
+            <GridListTile key={index} value={resident.id} onClick={() => handleClick(resident)}>
```

Comment:
So glad we got some help here: What worked was adding an index into the callback for `.map` and passing resident into handleClick.